packages/astro/src/core/render-context.ts

```diff
@@ -223,26 +222,25 @@ export class RenderContext {
 	 * API Context may be created multiple times per request, i18n data needs to be computed only once.
 	 * So, it is computed and saved here on creation of the first APIContext and reused for later ones.
 	 */
-	#i18nData?: Pick<APIContext, 'currentLocale' | 'preferredLocale' | 'preferredLocaleList'>;
-
-	get i18nData() {
-		if (this.#i18nData) return this.#i18nData;
-		const {
-			pipeline: { i18n },
-			request,
-			routeData,
-		} = this;
-		if (!i18n)
-			return {
-				currentLocale: undefined,
-				preferredLocale: undefined,
-				preferredLocaleList: undefined,
-			};
+	#currentLocale: APIContext['currentLocale'];
+	computeCurrentLocale() {
+		const { url, pipeline: { i18n }, routeData } = this;
+		if (!i18n) return;
 		const { defaultLocale, locales, strategy } = i18n;
-		return (this.#i18nData = {
-			currentLocale: computeCurrentLocale(routeData.route, locales, strategy, defaultLocale),
-			preferredLocale: computePreferredLocale(request, locales),
-			preferredLocaleList: computePreferredLocaleList(request, locales),
-		});
+		return this.#currentLocale ??= computeCurrentLocale(routeData.route, locales, strategy, defaultLocale);
```

Comment:
```suggestion
		return this.#currentLocale ??= computeCurrentLocale(routeData.pathname ?? routeData.route, locales, strategy, defaultLocale);
```

Reply:
Mmmh, I do recall using `.route`, and that caused of some issues... But if all tests pass, we should be fine. Weird though

Reply:
They would fail if this code was used for the Astro global as well.  The tests for `Astro.currentLocale` require `url.pathname`.